### PR TITLE
OTP 2 trip viewer repair

### PR DESCRIPTION
**Pull request by miles-grant-ibigroup**
This is the final fix for running OTP 1 and 2 in parallel.

The data returned from the trip endpoint of OTP 2 doesn't include some of the things the OTP 1 endpoint contained, and this causes it to crash in some cases.

This PR solves the issue in 2 ways. First, it makes a graphQL request to obtain the supplemental data. Second, it patches the trip viewer to not crash if this information is missing.

The extra GraphQL request is not getting that much information, and we may decide during the review process to not make this request and instead leave a comment noting that when we fully move to OTP-2 the full information can be retrieved.

### CODE REVIEW

**Comment by binh-dam-ibigroup** [comment]
Any reason the tests aren't showing?

**Comment by miles-grant-ibigroup** [comment]
I think it was because it was a merge commit rather than a "proper" commit. We can try pushing a dummy commit

File: lib/actions/api.js [hunk 1]
```diff
@@ -559,6 +560,56 @@ export function findTrip(params) {
     }
   )
 }
+export function findGraphQLTrip(params) {
+  // Attempt an otp-2 graphQL request for the same trip before failing
+  // since the otp-2 rest api doesn't return the needed fields
```
Comment (binh-dam-ibigroup) [inline lib/actions/api.js:565]: Convert this comment to JSDoc (insert blank line before JSDoc).

File: lib/components/viewers/trip-viewer.js [hunk 2]
```diff
@@ -1,29 +1,30 @@
...
 import moment from 'moment'
 import PropTypes from 'prop-types'
 import React, { Component } from 'react'
-import { Button, Label } from 'react-bootstrap'
-import { FormattedMessage, FormattedTime } from 'react-intl'
-import { connect } from 'react-redux'
 
+import { findGraphQLTrip, findTrip } from '../../actions/api'
+import { setLocation } from '../../actions/map'
+import { setViewedTrip } from '../../actions/ui'
```
Comment (binh-dam-ibigroup) [inline lib/components/viewers/trip-viewer.js:13]: Use the following syntax for importing actions that was adopted some time in 2020.
It prevents from accidentally calling an action without mapping it to dispatch:
```js
import * as apiActions from '../../actions/api'
...
const mapDispatchToProps = {
  findTrip: apiActions.findTrip
  ...
}

```

File: lib/components/viewers/trip-viewer.js [hunk 3]
```diff
@@ -32,91 +33,100 @@ class TripViewer extends Component {
...
+          {tripData &&
+            tripData.stops &&
+            tripData.stopTimes &&
             tripData.stops.map((stop, i) => {
               // determine whether to use special styling for first/last stop
               let stripMapLineClass = 'strip-map-line'
               if (i === 0) stripMapLineClass = 'strip-map-line-first'
-              else if (i === tripData.stops.length - 1) stripMapLineClass = 'strip-map-line-last'
+              else if (i === tripData.stops.length - 1)
+                stripMapLineClass = 'strip-map-line-last'
```
Comment (binh-dam-ibigroup) [inline lib/components/viewers/trip-viewer.js:120]: I would use brackets on multi-line if/else blocks.

File: lib/components/viewers/trip-viewer.js [hunk 4]
```diff
@@ -32,91 +33,100 @@ class TripViewer extends Component {
...
               let highlightClass
-              if (i === viewedTrip.fromIndex) highlightClass = 'strip-map-highlight-first'
-              else if (i > viewedTrip.fromIndex && i < viewedTrip.toIndex) highlightClass = 'strip-map-highlight'
-              else if (i === viewedTrip.toIndex) highlightClass = 'strip-map-highlight-last'
+              if (i === viewedTrip.fromIndex)
+                highlightClass = 'strip-map-highlight-first'
+              else if (i > viewedTrip.fromIndex && i < viewedTrip.toIndex)
+                highlightClass = 'strip-map-highlight'
+              else if (i === viewedTrip.toIndex)
+                highlightClass = 'strip-map-highlight-last'
```
Comment (binh-dam-ibigroup) [inline lib/components/viewers/trip-viewer.js:129]: I would use brackets here.

**Review by binh-dam-ibigroup — CHANGES_REQUESTED** [review]
Can you make one import syntax adjustment please? Seems to work fine otherwise. See other minor comments. too.

File: lib/actions/api.js (hunk 1 above)
Comment (miles-grant-ibigroup) [reply lib/actions/api.js]: Good call! Done in ca08debd338383fb327a3543108c92b3e269a740

File: lib/components/viewers/trip-viewer.js (hunk 2 above)
Comment (miles-grant-ibigroup) [reply lib/components/viewers/trip-viewer.js]: That's why we do that! Genius. Refactored in ca08debd338383fb327a3543108c92b3e269a740

File: lib/components/viewers/trip-viewer.js (hunk 3 above)
Comment (miles-grant-ibigroup) [reply lib/components/viewers/trip-viewer.js]: Good catch thanks! Prettier's doing

fixed in ca08debd338383fb327a3543108c92b3e269a740

File: lib/components/viewers/trip-viewer.js (hunk 4 above)
Comment (miles-grant-ibigroup) [reply lib/components/viewers/trip-viewer.js]: Good catch thanks! Prettier's doing

fixed in ca08deb

File: lib/components/viewers/trip-viewer.js [hunk 5]
```diff
@@ -1,29 +1,30 @@
...
 import moment from 'moment'
 import PropTypes from 'prop-types'
 import React, { Component } from 'react'
-import { Button, Label } from 'react-bootstrap'
-import { FormattedMessage, FormattedTime } from 'react-intl'
-import { connect } from 'react-redux'
 
+import * as mapActions from '../../actions/map'
+import * as uiActions from '../../actions/ui'
+import { findGraphQLTrip, findTrip } from '../../actions/api'
```
Comment (binh-dam-ibigroup) [inline lib/components/viewers/trip-viewer.js:13]: Do the same for API actions.

**Review by binh-dam-ibigroup — APPROVED** [review]
One more import refactor for API actions, please then I'm good.

File: lib/components/viewers/trip-viewer.js (hunk 5 above)
Comment (miles-grant-ibigroup) [reply lib/components/viewers/trip-viewer.js]: Slept right through that one! Thank you for catching

Extracted in 24d33b2cc7bc6c907163261e887bc141de279f1d

**Pull request merged**